### PR TITLE
use runc to unpause if needed

### LINKED ISSUE #1982: invoker shutdown does not take runc into account
## Steps to reproduce the issue:

1. install whisk  
2. create a new action and run it
3.  `docker stop invoker0`

## Provide the expected results and outputs:

```
Shutdown hook should unpause (runc resume) the container and remove it.
```

## Provide the actual results and outputs:

```
[2017-03-08T09:33:36.374Z] [WARN] [??] [ContainerPool] Shutdown hook activated.  Starting container shutdown
[2017-03-08T09:33:36.374Z] [INFO] [#sid_101] [ContainerUtils] [marker:invoker_docker.ps_start:23342545]
[2017-03-08T09:33:36.374Z] [INFO] [#sid_101] [SimpleExec] Running command: /usr/bin/docker ps --no-trunc -a
[2017-03-08T09:33:36.391Z] [INFO] [#sid_101] [SimpleExec] Done running command: /usr/bin/docker ps --no-trunc -a
[2017-03-08T09:33:36.391Z] [INFO] [#sid_101] [ContainerUtils] [marker:invoker_docker.ps_finish:23342564:19]
[2017-03-08T09:33:36.391Z] [INFO] [#sid_101] [ContainerPool] killStragglers now removing 1 leftover containers
[2017-03-08T09:33:36.392Z] [INFO] [#sid_101] [ContainerUtils] [marker:invoker_docker.unpause_start:23342565]
[2017-03-08T09:33:36.392Z] [INFO] [#sid_101] [SimpleExec] Running command: /usr/bin/docker unpause wsk0_177256_limeusibmcomactioninvokenodejs1488965337811001_20170308T092858748Z
[2017-03-08T09:33:36.407Z] [INFO] [#sid_101] [SimpleExec] Done running command: /usr/bin/docker unpause wsk0_177256_limeusibmcomactioninvokenodejs1488965337811001_20170308T092858748Z
[2017-03-08T09:33:36.407Z] [ERROR] [#sid_101] [ContainerUtils] stdout:  stderr: Error response from daemon: Container 480f96278f99592139a19e0cc4ba756f15d74c3052fa2d1524ebdf7edc840e70 is not paused [marker:invoker_docker.unpause_error:23342580:15]
[2017-03-08T09:33:36.407Z] [INFO] [#sid_101] [ContainerUtils] [marker:invoker_docker.rm_start:23342580]
[2017-03-08T09:33:36.407Z] [INFO] [#sid_101] [SimpleExec] Running command: /usr/bin/docker rm -f wsk0_177256_limeusibmcomactioninvokenodejs1488965337811001_20170308T092858748Z
```

## Additional information you deem important:

I suspect the culprit to be in the `killStragglers` function which is called during shutdown.
It directly calls `ContainerUtils.unpauseContainer` which implies `docker unpause`. It should however call `Container.unpause` on the container instance which knows whether to unpause using docker or runc.

### PR DESCRIPTION
**Pull request by domdom82**
fixes #1982

### CODE REVIEW

File: core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala [hunk 1]
```diff
@@ -778,7 +778,10 @@ class ContainerPool(
         val candidates = allContainers.filter { _.name.name.startsWith(actionContainerPrefix) }
         logging.info(this, s"killStragglers now removing ${candidates.length} leftover containers")
         candidates foreach { c =>
-            unpauseContainer(c.name)
+            useRunc match {
+              case true => RuncUtils.resume(c.id)
+              case false => unpauseContainer(c.name)
+            }
```
Comment (markusthoemmes) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: also known as `if(useRunc) RuncUtils.resume(c.id) else unpauseContainer(c.name)`

**Comment by domdom82** [comment]
got runc to work on mac just fine. I added the runc binary into the invoker container and let it use the `/run/runc` mount as before. only apparmor is not supported as docker-machine is not ubuntu-based.

enabled runc on local too, to get better test coverage with travis.

reverted my change to use both runc resume and docker unpause because they still get confused and docker unpause won't work if runc resume already unpaused the container.

in-between states (i.e. changing runc config between two invoker runs without proper `docker stop invoker0`) will be handled gracefully by the ansible playbook for invoker.

**Comment by domdom82** [comment]
PG4 33

**Comment by domdom82** [comment]
PG1 1245

**Comment by domdom82** [comment]
rebased on master

**Pull request merged**